metacat-main/src/main/java/com/netflix/metacat/main/services/impl/ConnectorTableServiceProxy.java

```diff
@@ -0,0 +1,229 @@
+     * @return table dto
+     */
+    @Cacheable(key = "'table.' + #name", condition = "#useCache")
+    public TableDto get(final QualifiedName name, final boolean useCache) {
+        final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
+        final ConnectorRequestContext connectorRequestContext = converterUtil.toConnectorContext(metacatRequestContext);
+        final ConnectorTableService service = connectorManager.getTableService(name);
+        return converterUtil.toTableDto(service.get(connectorRequestContext, name));
+    }
+
+    /**
+     * Calls the connector table service rename method.
+     * @param oldName old table name
+     * @param newName new table name
+     * @param isMView true, if the object is a view
+     */
+    @CacheEvict(key = "'table.' + #oldName")
+    public void rename(
+        final QualifiedName oldName,
+        final QualifiedName newName,
+        final boolean isMView
+    ) {
+        final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
+        final ConnectorTableService service = connectorManager.getTableService(oldName);
+
+        try {
+            log.info("Renaming {} {} to {}", isMView ? "view" : "table", oldName, newName);
+            final ConnectorRequestContext connectorRequestContext
+                = converterUtil.toConnectorContext(metacatRequestContext);
+            service.rename(connectorRequestContext, oldName, newName);
+        } catch (UnsupportedOperationException ignored) {
+        }
+    }
+
+    /**
+     * Calls the connector table service update method.
+     * @param name table name
+     * @param tableDto table object
+     */
+    @CacheEvict(key = "'table.' + #name")
+    public void update(final QualifiedName name, final TableDto tableDto) {
+        final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
+        final ConnectorTableService service = connectorManager.getTableService(name);
+        //Ignore if the operation is not supported, so that we can at least go ahead and save the user metadata
+        if (isTableInfoProvided(tableDto)) {
+            try {
+                log.info("Updating table {}", name);
+                final ConnectorRequestContext connectorRequestContext
+                    = converterUtil.toConnectorContext(metacatRequestContext);
+                service.update(connectorRequestContext, converterUtil.fromTableDto(tableDto));
+            } catch (UnsupportedOperationException ignored) {
+            }
+        }
+    }
+
+    private boolean isTableInfoProvided(final TableDto tableDto) {
+        boolean result = false;
+        if ((tableDto.getFields() != null && !tableDto.getFields().isEmpty())
+            || tableDto.getSerde() != null
+            || (tableDto.getMetadata() != null && !tableDto.getMetadata().isEmpty())
+            || tableDto.getAudit() != null) {
+            result = true;
```

Comment:
Why not check tableDto.getAudit().isEmpty() ?

Reply:
Because tableDto.getAudit() is not a list.